frontend/src/component/releases/ReleasePlanTemplate/EditReleasePlanTemplate.tsx

```diff
@@ -89,52 +78,78 @@ export const EditReleasePlanTemplate = () => {
+                />
+
+                <MilestoneList
+                    milestones={milestones}
+                    setAddStrategyOpen={setAddStrategyOpen}
+                    errors={errors}
+                    clearErrors={clearErrors}
+                />
+                <StyledAddMilestoneButton
+                    variant='text'
+                    color='primary'
+                    onClick={(e) => {
+                        e.preventDefault();
+                        setMilestones([
+                            ...milestones,
+                            {
+                                name: `Milestone ${milestones.length + 1}`,
+                                sortOrder: milestones.length,
+                            },
+                        ]);
+                    }}
+                >
+                    + Add milestone
+                </StyledAddMilestoneButton>
+                <StyledButtonContainer>
+                    <UpdateButton name='template' permission={ADMIN} />
+                    <StyledCancelButton onClick={handleCancel}>
+                        Cancel
+                    </StyledCancelButton>
+                </StyledButtonContainer>
+            </StyledForm>
+            <SidebarModal
+                label='Add strategy to template milestone'
+                onClose={onSidebarClose}
+                open={addStrategyOpen}
             >
-                <StyledForm onSubmit={handleSubmit}>
-                    <TemplateForm
-                        name={name}
-                        setName={setName}
-                        description={description}
-                        setDescription={setDescription}
-                        errors={errors}
-                        clearErrors={clearErrors}
+                <>
+                    <ReleasePlanTemplateAddStrategyForm
+                        onCancel={() => {
+                            setAddStrategyOpen(false);
+                        }}
                     />
-
-                    {template.milestones.map((milestone) => (
-                        <StyledMilestoneCard key={milestone.id}>
-                            <StyledMilestoneCardBody>
-                                <StyledMilestoneCardTitle>
-                                    {milestone.name}
-                                </StyledMilestoneCardTitle>
-                            </StyledMilestoneCardBody>
-                        </StyledMilestoneCard>
-                    ))}
-                    <StyledButtonContainer>
-                        <UpdateButton name='template' permission={ADMIN} />
-                        <StyledCancelButton onClick={handleCancel}>
-                            Cancel
-                        </StyledCancelButton>
-                    </StyledButtonContainer>
-                </StyledForm>
-            </FormTemplate>
-        </>
+                </>
+            </SidebarModal>
+        </FormTemplate>
```

Comment:
Wasn't sure how this would develop as I went along, and decided to keep them separate until I knew if I could merge them. I'll have another look now and see if I can merge them together